File: java/src/test/java/problems101_200/WordLadderTest.java
package problems101_200;

import java.util.Arrays;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class WordLadderTest extends SampledTest {
    @Test
    public void test() {
        for (WordLadder solution : WordLadder.values()) {
            for (WordLadderSample sample : getSamples(WordLadderSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.ladderLength(
                            sample.input.beginWord,
                            sample.input.endWord,
                            Arrays.asList(sample.input.wordList)
                        )
                    ).isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/WordLadderTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class WordLadderTest : SampledTest() {
    @Test
    fun test() =
        WordLadder.entries.forEach { solution ->
            getSamples(Array<WordLadderSample>::class.java).forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(
                        solution.ladderLength(
                            input.beginWord,
                            input.endWord,
                            input.wordList.toList(),
                        ),
                    ).isEqualTo(output)
            }
        }
}
